app/main/presenters/search_summary.py

```diff
@@ -103,17 +103,31 @@ def _group_request_filters(self, request_args, filter_groups):
         def _is_option(values):
             return (len(values) == 1) and (values[0] == u'true')
 
+        def _get_filter_recursive(key, value, filters):
+            """
+            Search filters (and their children, in the case of checkbox_tree questions) for a one
+            where filter[key] == value.
+            """
+            for filter in filters:
+                if filter[key] == value:
+                    return filter
+                children = filter.get('children')
+                if children:
+                    search_children = _get_filter_recursive(key, value, children)
+                    if search_children is not None:
+                        return search_children
+
```

Comment:
No actual reason for this to be a nested function (but then again the same is true for `_is_option` above)